cdci_data_analysis/analysis/parameters.py

```diff
@@ -964,4 +968,76 @@ def value(self, v):
         elif v in self._true_rep:
             self._value = True
         else:
-            raise RequestNotUnderstood(f'Wrong value for boolean parameter {self.name}')
\ No newline at end of file
+            raise RequestNotUnderstood(f'Wrong value for boolean parameter {self.name}')
+        
+class StructuredParameter(Parameter):
+    owl_uris = ("http://odahub.io/ontology#StructuredParameter")
+    
+    def __init__(self, value=None, name=None, schema=None):
+        
+        self.schema = schema
+        
+        if self.schema is None:
+            logger.warning("Parameter %s: Schema is not defined, will allow any structure.", name)
+            
+        super().__init__(value=value,
+                         name=name)
+    
+    def check_schema(self):
+        if self.schema is not None:
+            validate(self._value, self.schema)
+        
+    def additional_check(self):
+        # should raise AssertionError if wrong
+        pass
+    
+    def check_value(self):
+        try:
+            self.check_schema()
+            self.additional_check()
+        except (AssertionError, ValidationError):
+            raise RequestNotUnderstood('Wrong value of structured parameter %s', self.name)
+            
+
+class PhosphorosFiltersTable(StructuredParameter):
+    owl_uris = ('http://odahub.io/ontology#PhosphorosFiltersTable')
+    
+    def __init__(self, value=None, name=None):
+        
+        # TODO: this list should come from outside
+        filter_list = ["CFHT|MegaCam.g", "CFHT|MegaCam.gri", "CFHT|MegaCam.i", "CFHT|MegaCam.r", "CFHT|MegaCam.u", "CFHT|MegaCam.z",
+                       "CTIO|DECam.Y", "CTIO|DECam.g", "CTIO|DECam.i", "CTIO|DECam.r", "CTIO|DECam.u", "CTIO|DECam.z",
+                       "Euclid|NISP.H", "Euclid|NISP.J", "Euclid|NISP.Y", "Euclid|VIS.vis",
+                       "GAIA|GAIA3.G", "GAIA|GAIA3.Gbp", "GAIA|GAIA3.Grp",
+                       "GALEX|GALEX.FUV", "GALEX|GALEX.NUV",
+                       "LSST|LSST.g", "LSST|LSST.i", "LSST|LSST.r", "LSST|LSST.u", "LSST|LSST.y", "LSST|LSST.z",
+                       "PAN-STARRS|PS1.g", "PAN-STARRS|PS1.i", "PAN-STARRS|PS1.open", "PAN-STARRS|PS1.r", "PAN-STARRS|PS1.w", "PAN-STARRS|PS1.y", "PAN-STARRS|PS1.z",
+                       "VIRCAM|VISTA.H", "VIRCAM|VISTA.J", "VIRCAM|VISTA.Ks", "VIRCAM|VISTA.NB118", "VIRCAM|VISTA.NB980", "VIRCAM|VISTA.NB990", "VIRCAM|VISTA.Y", "VIRCAM|VISTA.Z",
+                       "SLOAN|SDSS.g", "SLOAN|SDSS.i", "SLOAN|SDSS.r", "SLOAN|SDSS.u", "SLOAN|SDSS.z",
+                       "Subaru|HSC.Y", "Subaru|HSC.g", "Subaru|HSC.i", "Subaru|HSC.r", "Subaru|HSC.z",
+                       "UKIRT|WFCAM.H", "UKIRT|WFCAM.J", "UKIRT|WFCAM.K", "UKIRT|WFCAM.Y", "UKIRT|WFCAM.Z",
+                       "WISE|WISE.W1", "WISE|WISE.W2", "WISE|WISE.W3", "WISE|WISE.W4"]
+        
+        schema = {"type": "object",
```

Comment:
ultimately we could even load this schema from a JSON document somewhere, purely based on the URI.
If there is no additional check, this would allow to avoid even having the class.
But it's ok at this stage in the code like that I think.